app/code/Magento/Inventory/Test/Acceptance/Test/AdminManageSourcesSearchByKeywordTest.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="../../../../../../vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/Test/etc/testSchema.xsd">
+    <test name="AdminManageSourcesSearchByKeywordTest">
+        <annotations>
+            <features value="Multi-Source Inventory"/>
+            <stories value="Search By Keyword in Grid"/>
+            <title value="Search by Keyword on the Manage Sources grid"/>
+            <description value="You should be able to Search by Keyword from the Manage Sources grid."/>
+            <testCaseId value="506"/>
+            <group value="msi"/>
+        </annotations>
+        <before>
+            <createData entity="TestSource1" stepKey="createSource1"/>
+            <createData entity="TestSource2" stepKey="createSource2"/>
+            <actionGroup ref="LoginAsAdmin" stepKey="loginAsAdmin"/>
+        </before>
+        <after>
+            <actionGroup ref="LogoutOfAdmin" stepKey="logoutOfAdmin"/>
+        </after>
+
+        <amOnPage url="{{ManageSources.url}}" stepKey="amOnTheSourcesListPage"/>
+        <waitForPageLoad stepKey="waitForPageLoad1"/>
+
+        <actionGroup ref="AdminGridSearchByKeyword" stepKey="searchByKeyword">
+            <argument name="value" value="$$createSource2.general[name]$$"/>
```

Comment:
In scenario described put only piece of source name and exactly number "4" to search by keyword field and verify that "Test Source 4" appears.

```
14. Put in "Search by keyword" field value "4"
15. verify that search returned "Test Source 4" in grid
16. Then Search by keyword worked properly in Manage Sources grid
```